File: app/src/main/java/com/ysl/mymvp/ui/fragment/BaseFragment.java
package com.ysl.mymvp.ui.fragment;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

/**
 * Created by ysl on 2017/7/26.
 */
public abstract class BaseFragment extends Fragment {
    public int kindFlag = 0;
    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {

        return setView(inflater, container);

    }

    public abstract View setView(LayoutInflater inflater, ViewGroup container);

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        afterView(view);
    }

    public abstract void afterView(View view);
    public abstract void setKindFlag(int kindFlag);
}

File: mykotlin/src/main/java/com/example/mykotlin/base/BaseFragment.kt
package com.example.mykotlin.base

import android.content.Context
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button

/**
 * Created by ysl on 2018/2/27.
 * 介绍:
 */

 abstract class BaseFragment : Fragment() {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return setView(inflater, container)
    }

    override fun onAttach(context: Context?) {
        super.onAttach(context)
    initData()
    }
    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        afterView()
    }

    open abstract fun setView(inflater: LayoutInflater?, container: ViewGroup?): View?;

    open abstract fun afterView()

    open abstract fun initData()
}